lightstep/instrumentation/runtime/defs.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+func expectRuntimeMetrics() *builtinDescriptor {
```

Comment:
With each new go version (and/or new minimum version), do these have to be updated?

Reply:
Whereas before, the registration was based on the `runtime/metrics.Kind` and there were version-dependent checks. Now, there is not a requirement that this defs file is up-to-date and it can be used on older versions safely. When new runtime/metrics become available, they will cause errors to print via `otel.Handle` but they are really only warnings. For example in go-1.20 there are two new metrics, if you upgraded from the 1.19 `defs.go` you'd see something like

```
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/gc/mark/assist:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/gc/mark/dedicated:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/gc/mark/idle:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/gc/pause:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/gc/total:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/idle:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/scavenge/assist:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/scavenge/background:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/scavenge/total:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/total:cpu-seconds
2022/12/20 17:13:11 unrecognized runtime/metrics name: /cpu/classes/user:cpu-seconds
```